examples/http/src/main/java/io/opentelemetry/example/http/HttpServer.java

```diff
@@ -40,40 +40,46 @@
 
 public class HttpServer {
 
-  private class HelloHandler implements HttpHandler {
+  private static class HelloHandler implements HttpHandler {
 
     @Override
-    public void handle(HttpExchange he) throws IOException {
+    public void handle(HttpExchange exchange) throws IOException {
       // Name convention for the Span is not yet defined.
       // See: https://github.com/open-telemetry/opentelemetry-specification/issues/270
-      Span.Builder spanBuilder = tracer.spanBuilder("/").setSpanKind(Span.Kind.SERVER);
-      Span span = null;
 
       // Extract the context from the HTTP request
-      Context ctx =
-          OpenTelemetry.getPropagators().getHttpTextFormat().extract(Context.current(), he, getter);
-      try (Scope scope = ContextUtils.withScopedContext(ctx)) {
-        // Build a span automatically using the received context
-        span = spanBuilder.startSpan();
+      Context context =
+          OpenTelemetry.getPropagators()
+              .getHttpTextFormat()
+              .extract(Context.current(), exchange, getter);
+
+      Span span =
+          tracer
+              .spanBuilder("/")
+              .setParent(TracingContextUtils.getSpan(context))
```

Comment:
is this the best way to get the parent?

Reply:
(Makes me remember I've been longing to remove the dust from some code locally, to have a `setContext(Context context)` overload).